packages/web/src/components/QueryHooksProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import type { DocumentNode } from 'graphql'
+
+export interface QueryHookOptions {
+  variables?: Record<string, any>
+}
+export interface OperationResult<TData = any> {
+  data?: TData | undefined
+  loading: boolean
+  error?: Error
+}
+
+export type MutationOperationResult<TData = any> = [
+  (options?: any) => Promise<TData>,
+  OperationResult<TData>
+]
+
+export interface QueryHooks {
+  mapUseQueryHook: (
```

Comment:
I'd probably just name this `useQuery`. Otherwise I think it should be `mappedUseQueryHook` or possibly `registeredUseQueryHook`